skrub/_interpolation_join.py

```diff
@@ -0,0 +1,328 @@
+
+    suffix : str
+        Suffix to append to the ``right_table``'s column names. You can use it
+        to avoid duplicate column names in the join.
+
+    regressor : scikit-learn regressor
+        Model used to predict the numerical columns of ``right_table``.
+
+    classifier : scikit-learn classifier
+        Model used to predict the categorical (string) columns of
+        ``right_table``.
+
+    vectorizer : scikit-learn transformer that can operate on a DataFrame or None
+        If provided, it is used to transform the feature columns before passing
+        them to the scikit-learn estimators. This is useful if we are joining
+        on columns that cannot be used directly, such as timestamps or strings
+        representing high-cardinality categories. If None, no transformation is
+        applied.
+
+    n_jobs : int
+        Number of joblib workers to use Depending on the estimators used and
+        the contents of ``right_table``, several estimators may need to be
+        fitted -- for example one for continuous outputs (regressor) and one
+        for categorical outputs (classifier), or one for each column when the
+        provided estimators do not support multi-output tasks. Fitting and
+        querying these estimators can be done in parallel.
+
+    Attributes
+    ----------
+    vectorizer_ : scikit-learn transformer
+        The transformer used to vectorize the feature columns.
+
+    estimators_ : list of dicts
+        The estimators used to infer values to be joined. Each entry in this
+        list is a dictionary with keys ``"estimator"`` (the fitted estimator) and
+        ``"columns"`` (the list of columns that it is trained to predict).
+
+    See Also
+    --------
+    Joiner :
+        Works in a similar way but instead of inferring values, picks the
+        closest row from the right table.
+
+    Examples
+    --------
+    >>> buildings
+       latitude  longitude  n_stories
+    0       1.0        1.0          3
+    1       2.0        2.0          7
+
+    >>> annual_avg_temp
+       latitude  longitude  avg_temp
+    0       1.2        0.8      10.0
+    1       0.9        1.1      11.0
+    2       1.9        1.8      15.0
+    3       1.7        1.8      16.0
+    4       5.0        5.0      20.0
+
+    >>> InterpolationJoin(
+    ...     annual_avg_temp,
+    ...     on=["latitude", "longitude"],
+    ...     regressor=KNeighborsRegressor(2),
+    ... ).fit_transform(buildings)
+       latitude  longitude  n_stories  avg_temp
+    0       1.0        1.0          3      10.5
+    1       2.0        2.0          7      15.5
+    """
+
+    def __init__(
+        self,
+        right_table,
+        *,
+        left_on=None,
+        right_on=None,
+        on=None,
+        suffix="",
+        regressor=ensemble.HistGradientBoostingRegressor(),
+        classifier=ensemble.HistGradientBoostingClassifier(),
+        vectorizer=TableVectorizer(),
```

Comment:
in any case we have to make sure to clone them before modifying them, because we don't want to modify the user-provided estimators (if any), so mutability shouldn't be an issue. for clarity, I would say putting them here directly avoids the need to look up what `None` means, so I'm not sure `None` is clearer?

Reply:
or always vectorize the key, as is done in the Joiner